core/templates/core/layout.html

```diff
@@ -0,0 +1,257 @@
+  <script src="{% static 'assets/js/accordions.js' %}"></script>
+
+  <script>
+    // This function is called to clear session storage when user logout
+    function logout() {
+      sessionStorage.removeItem("access")
+      sessionStorage.removeItem("refresh")
+      sessionStorage.removeItem("user_id")
+      window.location.href = "{% url 'authentication:login-view' %}"
+    }
+
+    function isLogin() {
+      const Bearer_access = sessionStorage.access
+      const Bearer_refresh = sessionStorage.refresh
+      const user_id = sessionStorage.user_id
+      const header = new Headers({
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer ' + Bearer_access
+      });
+      if ((Bearer_access != undefined) & (Bearer_refresh != undefined) & (user_id != undefined)) {
+        token = {
+          "Bearer_access": Bearer_access,
+          "Bearer_refresh": Bearer_refresh,
+          "user_id": user_id,
+          "header": header
+        }
+        return token;
+      }
+      else {
+        return false
+      }
+    }
+
+    // Gettin the name of the user
+    async function getUser(token) {
+
+      const url = "/auth/user/" + token.user_id
+      let response = await fetch(url, {
+        method: 'GET',
+        headers: token.header
+      });
+
+      data = await response.json()
+      if (response.ok) {
+        document.getElementById("login").classList.add("d-none");
+        document.getElementById("logout").classList.remove("d-none");
+        document.getElementById("user-name").innerHTML = "Mr. " + data.username
+      }
+    }
+
+    // Get the number of products in Cat if the user is loged in
+    async function getCartCount(token) {
+
+      // url for cart-view. We are counting the number of products in Cart of a user 
+      url = "/api/cart/view/" + token.user_id + "/"
+
+      let response = await fetch(url, {
+        method: 'GET',
+        headers: token.header
+      });
+      data = await response.json();
+      if (response.ok) {
+        let counter = 0
+        // get count of products if products are in the cart
+        for ([key, value] of Object.entries(data)) {
+          counter = counter + 1
+        }
+        // Update the cart value and link of cart
+        //debugger
+        url = "/cart/" + token.user_id
+        document.getElementById("cart_count").setAttribute("value", counter)
+        document.getElementById("cart-count-link").href = url
+      }
+    }
+
+    function userProfile(){
+
+    }
+
+    // Document ready function
+    $(document).ready(function () {
+      // Getting the name of user and number of product in the cart if user is logedin
+      //debugger
+      token = isLogin()
+      if (token) {
+        getUser(token)
+
+        getCartCount(token)
+      }
+
+    });
+
+  </script>
+  <script language="text/Javascript">
+    cleared[0] = cleared[1] = cleared[2] = 0; //set a cleared flag for each field
```

Comment:
not sure if I'm able to understand this piece of code

Reply:
Removed this piece of code. This was useless.